Source/WebCore/loader/ThreadableLoaderClient.h

```diff
@@ -42,7 +43,7 @@ class ResourceResponse;
 class ResourceTiming;
 class SharedBuffer;
 
-class ThreadableLoaderClient : public CanMakeThreadSafeCheckedPtr {
+class ThreadableLoaderClient : public CanMakeWeakPtr<ThreadableLoaderClient>, public CanMakeThreadSafeCheckedPtr {
```

Comment:
Hm... is it really safe to use `CanMakeWeakPtr` here? Shouldn't we be using thread safe weakptr?

Reply:
Thread safe WeakPtr requires ThreadSafeRefCounted too, a little awkward for a client class I think.

WeakPtr is thread-safe by default. The only requirement is that we have to deref it on the thread the object gets constructed/destroyed and we have assertions making sure this holds. EWS is green here in debug.